src/Geometry/NewestVertexBisection.jl

```diff
@@ -0,0 +1,308 @@
+        midpoint = _get_midpoint(node[[elem2, elem3], :])
+        node = push!(node, midpoint)
+        t = dualedge[elem[t][3], elem[t][2]]
+        # There is no dual edge here, go to next triangle index
+        if t == 0
+          need_to_mark = false
+        end
+      end
+    end
+  end
+  node, marker
+end
+
+function _divide!(elem::AbstractVector, t::T, p::AbstractVector{T}) where {T <: Integer}
+  new_row = [p[4], p[3], p[1]]
+  update_row = [p[4], p[1], p[2]]
+  push!(elem, new_row)
+  elem[t] = update_row
+  #elem = append_tables_globally(elem, Table([new_row]))
+  #for j = 1:3
+  #  elem.data[elem.ptrs[t] - 1 + j] = update_row[j]
+  #end
+  #elem[t][:] = [p[4] p[1] p[2]]
+  elem
+end
+
+function _bisect(
+    d2p::SparseMatrixCSC{T, T},
+    elem::AbstractVector,
+    marker::AbstractVector{T},
+    NT::T,
+  ) where {T<:Integer}
+  for t = UnitRange{T}(1:NT)
+    base = d2p[elem[t][2], elem[t][3]]
+    if (marker[base] > 0)
+      p = vcat(elem[t][:], marker[base])
+      elem = _divide!(elem, t, p)
+      left = d2p[p[1], p[2]]
+      right = d2p[p[3], p[1]]
+      if (marker[right] > 0)
+        cur_size::T = size(elem, 1)
+        elem = _divide!(elem, cur_size, [p[4], p[3], p[1], marker[right]])
+      end
+      if (marker[left] > 0)
+        elem = _divide!(elem, t, [p[4], p[1], p[2], marker[left]])
+      end
+    end
+  end
+  elem
+end
+
+function _build_edges(elem::AbstractVector{<:AbstractVector{T}}) where {T <: Integer}
+  edge = zeros(T, 3*length(elem), 2)
+  for t = 1:length(elem)
+    off = 3*(t-1)
+    edge[off+1,:] = [elem[t][1] elem[t][2]]
+    edge[off+2, :] = [elem[t][1] elem[t][3]]
+    edge[off+3, :] = [elem[t][2] elem[t][3]]
+  end
+  unique(sort!(edge, dims = 2), dims = 1)
+end
+
+function _build_directed_dualedge(elem::AbstractVector{<:AbstractVector{T}}, N::T, NT::T) where {T <: Integer}
+  dualedge = spzeros(T, N, N)
+  for t = 1:NT
+    dualedge[elem[t][1], elem[t][2]] = t
+    dualedge[elem[t][2], elem[t][3]] = t
+    dualedge[elem[t][3], elem[t][1]] = t
+  end
+  dualedge
+end
+
+function _dual_to_primal(edge::Matrix{T}, NE::T, N::T) where {T <: Integer}
+  d2p = spzeros(T, T, N, N)
+  for k = 1:NE
+    i = edge[k, 1]
+    j = edge[k, 2]
+    d2p[i, j] = k
+    d2p[j, i] = k
+  end
+  d2p
+end
+
+function is_against_top(face::Table{<:Integer}, top::GridTopology, d::Integer)
+  face_vec = sort.([face[i][:] for i = 1:size(face, 1)])
+  face_top = sort.(get_faces(top, d, 0))
+  issetequal_bitvec = issetequal(face_vec, face_top)
+  all(issetequal_bitvec)
+end
+
+function is_against_top(face::Matrix{<:Integer}, top::GridTopology, d::Integer)
+  face_vec = sort.([face[i, :] for i = 1:size(face, 1)])
+  face_top = sort.(get_faces(top, d, 0))
+  issetequal_bitvec = issetequal(face_vec, face_top)
+  all(issetequal_bitvec)
+end
+
+function _get_midpoint(ngon::AbstractArray{<:VectorValue})
+  sum(ngon) / length(ngon)
+end
+
+function Base.atan(v::VectorValue{2,T}) where {T<:AbstractFloat}
+  atan(v[2], v[1])
```

Comment:
Do not add this function (See bellow).

BTW, is this correct, v[2],v[1]?  Perhaps v[1],v[2] ?

Reply:
@fverdugo I was going off of [this definition](https://docs.julialang.org/en/v1/base/math/#Base.atan-Tuple%7BNumber%7D) with y = v[2], x = v[1] is that correct to assume?